### DISCUSSION TITLE
DEST path does not exist

### DISCUSSION THREAD

**Discussion by tamesoud** [RhetTbull/osxphotos#1406] [Q&A]
I've entered this a million times, can't figure out what I'm messing up:

osxphotos export --db /Volumes/TCmatic/Fotobackup24mar2021totUSA2022/FotosLibrary.photoslibrary --export-by-date --from-date 2020-01-01 --to-date 2020-03-01 --exiftool --album-keyword --person-keyword “/TCmatic/PhotosExport/“

I keep getting: 

Error: Invalid value for 'DEST': Path '“/TCmatic/PhotosExport/“' does not exist.

I've tried multiple variations without succes. TCmatic is an external harddrive. 

When I enter ls /Volumes it shows like this : GoogleDrive	Macintosh SD	TCmatic

so I'm thinking it should be there.

Any help would be appreciated. Don't shoot me, I'm a complete nutcase at this.

**Comment by RhetTbull** [comment]
If it's an external drive, you need to include `/Volumes` as part of the path. That's the mount point for external disks. In the Finder they show up just as the drive name but to the underlying operating system they are all mounted under `/Volumes`

Try: 

`osxphotos export --db /Volumes/TCmatic/Fotobackup24mar2021totUSA2022/FotosLibrary.photoslibrary --export-by-date --from-date 2020-01-01 --to-date 2020-03-01 --exiftool --album-keyword --person-keyword “/Volumes/TCmatic/PhotosExport/“`

**Reply by RhetTbull** [reply]
By the way, it looks like your Photos library and your export destination are both on an external drive. In that case, I would also use `--ramdb` as this will significantly boost export speed.  OSXPhotos maintains a database in the export destination to track which files have been exported, etc. but reading/writing to the external drive during export can be slow so `--ramdb` keeps the database in memory until the export is finished then writes it to disk.

**Comment by tamesoud** [comment]
Error: Invalid value for 'DEST': Path '“/Volumes/TCmatic/PhotosExport/“'
does not exist.


same shit I'm afraid...

Op di 13 feb 2024 om 21:13 schreef Rhet Turnbull ***@***.***>:

**Reply by RhetTbull** [reply]
Try `ls /Volumes/TCmatic/PhotosExport/`

Does that work?

Does the PhotosExport directory exist? osxphotos won't create the directory if it doesn't already exist.

**Comment by tamesoud** [comment]
Yes, the directory exists. I've also tried other directories on the
computer's internal harddrive.

ls /Volumes/TCmatic/PhotosExport/ does not generate a error message, if I
change it into something that does not exist, I do get the error:

Nicoles-MacBook-Pro:~ nicoleoud$ ls /Volumes/TCmatic/PhotosExport/

Nicoles-MacBook-Pro:~ nicoleoud$ ls /Volumes/TCmatic/PhotosExpo

ls: /Volumes/TCmatic/PhotosExpo: No such file or directory

Nicoles-MacBook-Pro:~ nicoleoud$



Op di 13 feb 2024 om 22:33 schreef Rhet Turnbull ***@***.***>:

**Reply by tamesoud** [reply]
I know a bit of Python, so i'll try via pip install. will post result here

**Reply by tamesoud** [reply]
Found it. It was notes in apple which replaced the straight quotation marks with curly onces....
I used it to edit the command line cmd.

You can fix it when making a new note, under edit, substitutions in the notes app on mac

Took a night without sleep, sorry for the trouble

**Reply by jaymis** [reply]
Thanks for posting this @tamesoud - I had the same issue, I was building my terminal command in Notes, with the hope to reuse it later. Ended up doing all kinds of weird stuff to try and fix my issue, including renaming my network volume so it didn't have spaces!

**Reply by RhetTbull** [reply]
It is possible for osxphotos to detect this (the use of curly quotes and report a helpful error message. I'll open an issue to add this.

**Reply by jaymis** [reply]
Thanks @RhetTbull, I've disabled smart quotes in Notes so it won't happen again, but I'm sure that will help someone else in the future!

**Reply by RhetTbull** [reply]
@jaymis Yeah, disabling all autocorrect including "smart quotes" is one of the first things I do on a new device so I forget this is a thing!

**Reply by jaymis** [reply]
I have only just started using Apple Notes this week, so it has come as quite a shock! Previously I was using Evernote for many years, but got sick of their constant nagging popups.

**Reply by RhetTbull** [reply]
I've got this implemented as follows.  Will need to add tests then will include this in the next release:

```
osxphotos export  --library ~/Pictures/Test-13.0.0.photoslibrary  “/Users/rhet/Desktop/export”
Usage: osxphotos export [OPTIONS] DEST
Try 'osxphotos export -h' for help.

Error: Invalid value for 'DEST': Path “/Users/rhet/Desktop/export” contains smart quotes. Please check the path and try again. If you copied this path from an app like Notes or TextEdit, try disabling smart quotes in the app.
```
